src/app/Modules/auth/pages/login/login.component.spec.ts

```diff
@@ -30,4 +29,35 @@ describe('LoginComponent', () => {
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should run #ngOnInit()', async () => {
+    component.form = component.formBuilder.group({
+      email: component.email,
+      password: component.password,
+    });
+    spyOn(component.formBuilder, 'group');
+    component.ngOnInit();
+    // expect(component.formBuilder.group).toHaveBeenCalled();
+  });
+
+  it('should run #getErrorMessage()', async () => {
```

Comment:
another thing that I noticed is that all the tests are marked as `async`, while the code looks synchronous, what is the reason for that?